packages/mermaid/src/Diagram.ts

```diff
@@ -50,7 +50,9 @@ export class Diagram {
     this.parser.parse = (text: string) =>
       originalParse(cleanupComments(extractFrontMatter(text, this.db, configApi.addDirective)));
 
-    this.parser.parser.yy = this.db;
+    if (this.parser.parser !== undefined) {
+      this.parser.parser.yy = this.db;
+    }
```

Comment:
I'm a still a bit confused why we're doing this. Can you add a comment?

My current guess is that if `this.parser.parser` is `undefined`, then it's the new Langium parser, but I'm not 100% sure.

Reply:
Yes. The `parser.parser` is only present in JISON. For any other parsers, like external diagrams, that would be undefined. So, we'll only set if present.